src/P2P/TaskQueue.hs

```diff
@@ -171,7 +171,7 @@ session_ limit q logFun env = mask $ \restore -> do
     let go = tryReadIVar (_taskResult task) >>= \case
             Nothing -> do
                 logg task Debug "run task"
-                flip catchAllSynchronous (retry task) $ restore $ do
+                handle (retry task) $ restore $ do
```

Comment:
This is only safe if we assume that all relevant exceptions arise from computations in `IO`, which probably is the case.